### ISSUE TITLE
Buttons in array editor are not setup correctly if array editor is initially collapsed

### ISSUE THREAD

**Issue by ThomasKiesel** [json-editor/json-editor#1453]
### General information

- json-editor version: 2.13.1

### Expected behavior

If the scheme contains an array-typed property which shall be collapsed initially, and the settings "display array delete all rows buttons" and "display array delete last row buttons" are set to true, the buttons must not be displayed when the array list is expanded.

### Actual behavior

The buttons are displayed when the array is expanded, in opposite to the settings "display array delete all rows buttons" and "display array delete last row buttons".

### Steps to reproduce the behavior

Direct link to example: https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUBXRBTA+vAhgLa75RQBOMARulFsigGb4A22ANCBOWIdDlSyMw5XABMsLLHVTM2WTpnFD86FlBwA3VunqzWHEIXwAPHBOgALVAAZONMghyIopXITATUARk4BjfGxnLCQYWE1cbkgscigAT2D8cj9rJgMFEBg/BBZqVBBheFcAd3peLABWEE4wKgArLD8NFnw4sFp8+BFjFmqQRBSsY1RQeIgsfNqGpr6o8diYPTQuaQZRuPH8pPJWvtgoKXyABVW+sKG1kDGJlHB6xqg9sMPbk8fOOZjYJfXN5ev8i5KPAAOYgAC+nAIxBGVw2N36FBgoIh4Mh4GgcCQsOyLBaEGwYlQFF0aLRnEQljAxRwMR45AYmSKMXwTSxe0sQwRVDAYCgQPwEAALH1RABHdAwURiAQJCTMNSPNLyThiGCICAtBLiyXSnA5OL6FX9Kk0yAaZFGwxdHD4MRq2AIVg4T4LJZyQz4OqmK0ZNWIfBUKS0h04OqIBC+1XqwPB3H47BRkD+2ORHjzb4MD1+mNBkjkHYJO1E5WGFN520F1o4UQiCTkJPl4PbasSKQyUsZEKpyuFnDZCCGzucFsJbJFHhsHBQSCN3PNqtyyTSEh4mvUhgknMBiuj8zLug4FoudfFYnkUngoA==

```javascript
{
  "type": "object",
  "properties": {
    "pets": {
      "type": "array",
      "title": "Pets",
      "items": {
        "type": "object",
        "title": "Pet",
        "properties": {
          "type": {
            "type": "string"
          },
          "name": {
            "type": "string"
          }
        }
      },
      "options": {
        "collapsed": true
      }
    }
  }
}
```

Settings: 
```
disable_array_delete_last_row: true,
disable_array_delete_all_rows: true
```
When the node "Pets" is expanded, the unwanted buttons are displayed.

### Cause

Most likely, the problem is caused in src/editors/array.js, l. 457
```
if (!this.collapsed && this.setupButtons(minItems)) { ... } 
```
If `this.collapsed === false`, the method `this.setupButtons(minItems)` is not executed. Possibly, exchanging both conditions is sufficient to fix the bug.

**Comment by ThomasKiesel** [comment]
Unfortunately, the fix is incomplete: it does not work if the array is initially empty. I am still working on this problem.

**Comment by ThomasKiesel** [comment]
**Update on my last comment**

I have created a new issue for the remaining problems: #1461. We are going to provide a fix for the new issue soon.
From my point of view, the current issue may be seen as closed.

**Comment by germanbisurgi** [comment]
Fixed in #1454
